CMakeLists.txt

```diff
@@ -19,6 +19,24 @@ endif (POLICY CMP0023)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules/")
 
+# find out which platform we are building on
+if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+  message(STATUS "Building for Linux.")
+  set(LINUX ON)
+  set(UNIX ON)
+  FIND_PACKAGE(Threads)
+elseif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+  message(STATUS "Building for FreeBSD.")
+  set(FREEBSD ON)
+  set(UNIX ON)
+  FIND_PACKAGE(Threads)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  message(STATUS "Building for Darwin.")
+  set(OperatingSystem "Mac OS X")
```

Comment:
no need to set this, cmake have `APPLE`. see https://cmake.org/cmake/help/v3.0/variable/APPLE.html